core/ledger/kvledger/txmgmt/statedb/statecouchdb/commit_handling_test.go

```diff
@@ -150,3 +150,132 @@ func TestBuildCommitters(t *testing.T) {
 	committer, err = db.buildCommitters(badBatch)
 	assert.EqualError(t, err, "nil version not supported")
 }
+
+func TestExecuteCommitter(t *testing.T) {
+	env := testEnv
+	env.init(t, &statedb.Cache{})
+	defer env.cleanup()
+
+	versionedDB, err := testEnv.DBProvider.GetDBHandle("test-execute-committer")
+	assert.NoError(t, err)
+	db := versionedDB.(*VersionedDB)
+
+	committerDB, err := db.getNamespaceDBHandle("ns")
+	assert.NoError(t, err)
+	couchDocKey1, err := keyValToCouchDoc(&keyValue{
+		key:            "key1",
+		revision:       "",
+		VersionedValue: &statedb.VersionedValue{Value: []byte("value1"), Metadata: nil, Version: version.NewHeight(1, 1)},
+	})
+	assert.NoError(t, err)
+	couchDocKey2, err := keyValToCouchDoc(&keyValue{
+		key:            "key2",
+		revision:       "",
+		VersionedValue: &statedb.VersionedValue{Value: nil, Metadata: nil, Version: version.NewHeight(1, 1)},
+	})
+	assert.NoError(t, err)
+
+	committers := []*committer{
+		{
+			db:             committerDB,
+			batchUpdateMap: map[string]*batchableDocument{"key1": {CouchDoc: *couchDocKey1}},
+			namespace:      "ns",
+			cacheKVs:       make(statedb.CacheKVs),
+			cacheEnabled:   true,
+		},
+		{
+			db:             committerDB,
+			batchUpdateMap: map[string]*batchableDocument{"key2": {CouchDoc: *couchDocKey2}},
+			namespace:      "ns",
+			cacheKVs:       make(statedb.CacheKVs),
+			cacheEnabled:   true,
+		},
+	}
+
+	err = db.executeCommitter(committers)
+	assert.NoError(t, err)
+	vv, err := db.GetState("ns", "key1")
+	assert.NoError(t, err)
+	assert.Equal(t, vv.Value, []byte("value1"))
+	assert.Equal(t, vv.Version, version.NewHeight(1, 1))
+
+	committers = []*committer{
+		{
+			db:             committerDB,
+			batchUpdateMap: map[string]*batchableDocument{},
+			namespace:      "ns",
+			cacheKVs:       make(statedb.CacheKVs),
+			cacheEnabled:   true,
+		},
+	}
+	err = db.executeCommitter(committers)
+	assert.EqualError(t, err, "error handling CouchDB request. Error:bad_request,  Status Code:400,  Reason:`docs` parameter must be an array.")
+}
+
+func TestCommitUpdates(t *testing.T) {
```

Comment:
One of the tasks of _commitUpdate()_ function is to update the _revision_ present in the cache. This is done by maintaining the newly created _revision_ in the _cacheKVs_ map present in the _committer_ object. It would be good to add a test to check whether the _revision_ is being updated correctly. We can do the following to check that:
1. add the key and the empty _revision_ (as the key is new) to the _cacheKVs_ map in the _committer_ object.
2. once the committer completes its execution, it would update the _cacheKVs_ with the assigned _revision_ for the key. 
3. check whether the _revision_ has been changed in the _cacheKVs_ map.

Reply:
thanks for the review. new test cases has been added.